File: TopCoder/Math/SRM610Div2.py
class DivideByZero:
	def CountNumbers(self,numbers):
		s = set(numbers)
		old = 0
		#repeat as long as there are new numbers in the list:
		while len(s) > old:
			old = len(s)
			# find a good pair (a,b) 
			for a in list(s):
				for b in list(s):
					if a > b and a/b not in s:
						s.add(a/b)
	return len(s)

File: TopCoder/Math/SRM610Div2.cpp
#include <iostream>
#include <vector>
#include <set>
using namespace std;

int CountNumbers(vector<int> numbers)
{
    int old = 0;
    set<int> paper(numbers.begin(), numbers.end());        
    while (paper.size() > old) {
        old = paper.size();
        //不能直接对set用[]
        for (int a: paper) {
            for (int b: paper) {
                if ( (a > b) && (paper.count(a/b) == 0) ) {
                    paper.insert(a / b);
                }
            }
        }
    }
    return paper.size();
}
